docs/docs/using.md

```diff
@@ -0,0 +1,151 @@
+# Using JSpecify Annotations
+
+## Depending on the annotations
+
+The annotations are available from Maven Central as
+[`org.jspecify:jspecify:1.0.0`](https://repo1.maven.org/maven2/org/jspecify/jspecify/1.0.0/).
+The annotations themselves are in the `org.jspecify.annotations` package.
+
+Below are snippets you can use to add a dependency in Maven, Gradle, or Bazel.
+
+Regardless of the build system you use, avoid configuring your build to hide
+JSpecify annotations from your users. We also recommend including the
+annotations at runtime, and we have kept the JSpecify jar small to reduce the
+cost of doing so. Each build tool supports different mechanisms that will hide
+the annotation declarations, and we recommend against them in the tool-specific
+guidance below.
+
+#### Maven
+
+```xml
+<dependency>
+  <groupId>org.jspecify</groupId>
+  <artifactId>jspecify</artifactId>
+  <version>1.0.0</version>
+</dependency>
+```
+
+Avoid using [provided] or [optional] scope.
+
+#### Gradle
+
+```groovy
+dependencies {
+  api("org.jspecify:jspecify:1.0.0")
+}
+```
+
+Avoid using [implementation] or (if your users might need the annotations at
+runtime) [compileOnlyApi] configurations.
+
+#### Bazel
+
+```python
+maven_jar(
+    name = "jspecify",
+    artifact = "org.jspecify:jspecify:1.0.0",
+    sha256 = "TODO: fill this in after we build the jar",
+)
+```
+
+## Switching to JSpecify nullness annotations
+
+### If your code has no nullness annotations
+
+If your code doesn't use nullness annotations yet, there's no time like the
+present! We have a high-level [strategy](applying) we recommend.
+
+### If your code already uses JSR-305 annotations
+
+Before migrating from the JSR-305 annotations to JSpecify annotations, see if
+the caveats about [Kotlin][Kotlin-caveats] or
+[annotation processors](whether#annotation-processors) apply to your situation.
+
+Migrating from JSR-305 annotations primarily entails changing imports, updating
+annotation names and locations, and addressing build errors. JSpecify's
+annotations are type-use annotations, which impose additional restrictions on
+where they are placed. In some cases, these restrictions can make their
+placement incompatible with the placement of existing JSR-305 annotations. We
+recommend taking the following migration steps:
+
+1.  Update imports to use JSpecify annotations.
+
+1.  Check *all* annotations on array types. Any code that currently has a type
+    like `@Nullable Object[]` must change to `Object @Nullable []`. This change
+    is required by the syntax of [type-use annotations]. If you do *not* make
+    this change, your code will change from meaning "a nullable array of
+    objects" to "an array of nullable objects."
+
+1.  Rebuild your project, making changes as needed to correct build errors.
+    Build errors can arise from restrictions on placing type-use annotations.
+    For example, if you see an error about an annotation on a "scoping
```

Comment:
In hindsight, I find the phrasing here a little weird. "Scoping construct" errors might be the _only_ such errors, not just an "example" of them. The part of this sentence that it is definitely an "example" is the part about `Map.Entry`.

> For example, if you see an error about an annotation on a "scoping construct" in `@Nullable Map.Entry<K, V>`, then you must change it to `Map.@Nullable Entry<K, V>`.